Link new PyTorch Contributing Guidelines from CONTRIBUTING.md

zou3519 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #103986

We wrote some new Contributing Guidelines that guide a contributor
through the lifecycle of a Pull Request to PyTorch.

We've gotten some positive feedback from early adopters so we are now
adding it as the go-to link in CONTRIBUTING.md and the PyTorch Wiki.

Note that there are older contributing guidelines over at
https://github.com/pytorch/pytorch/blob/main/docs/source/community/contribution_guide.rst
The new Contributing Guidelines doc is targeted towards guiding a user
through submitting and merging a Pull Request to pytorch; the existing
guidelines are more of a high-level overview. We should rationalize these
at some point, but I left the resources for now.

zou3519 committed:
Link new PyTorch Contributing Guidelines from CONTRIBUTING.md

We wrote some new Contributing Guidelines that guide a contributor
through the lifecycle of a Pull Request to PyTorch.

We've gotten some positive feedback from early adopters so we are now
adding it as the go-to link in CONTRIBUTING.md and the PyTorch Wiki.

Note that there are older contributing guidelines over at
https://github.com/pytorch/pytorch/blob/main/docs/source/community/contribution_guide.rst
The new Contributing Guidelines doc is targeted towards guiding a user
through submitting and merging a Pull Request to pytorch; the existing
guidelines are more of a high-level overview. We should rationalize these
at some point, but I left the resources for now.

[ghstack-poisoned]

albanD commented on CONTRIBUTING.md:
```diff
@@ -57,24 +57,13 @@
-This document covers some of the more technical aspects of contributing
-to PyTorch.  For more non-technical guidance about how to contribute to
-PyTorch, see the [Contributing Guide](docs/source/community/contribution_guide.rst).
+Thank you for your interest in contributing to PyTorch!
```
I would put this intro before the table of content to make sure it is the first thing people read!

zou3519 replied:
Good point! The TOC is long

albanD reviewed (approved):
small nit about positioning but SGTM otherwise

zou3519 committed:
Update on "Link new PyTorch Contributing Guidelines from CONTRIBUTING.md"


We wrote some new Contributing Guidelines that guide a contributor
through the lifecycle of a Pull Request to PyTorch.

We've gotten some positive feedback from early adopters so we are now
adding it as the go-to link in CONTRIBUTING.md and the PyTorch Wiki.

Note that there are older contributing guidelines over at
https://github.com/pytorch/pytorch/blob/main/docs/source/community/contribution_guide.rst
The new Contributing Guidelines doc is targeted towards guiding a user
through submitting and merging a Pull Request to pytorch; the existing
guidelines are more of a high-level overview. We should rationalize these
at some point, but I left the resources for now.

[ghstack-poisoned]

zou3519 committed:
Update on "Link new PyTorch Contributing Guidelines from CONTRIBUTING.md"


We wrote some new Contributing Guidelines that guide a contributor
through the lifecycle of a Pull Request to PyTorch.

We've gotten some positive feedback from early adopters so we are now
adding it as the go-to link in CONTRIBUTING.md and the PyTorch Wiki.

Note that there are older contributing guidelines over at
https://github.com/pytorch/pytorch/blob/main/docs/source/community/contribution_guide.rst
The new Contributing Guidelines doc is targeted towards guiding a user
through submitting and merging a Pull Request to pytorch; the existing
guidelines are more of a high-level overview. We should rationalize these
at some point, but I left the resources for now.

[ghstack-poisoned]

albanD reviewed (approved):
Thanks!

zou3519 commented:
@pytorchbot merge
